CIPs/cip-granda_mento.md

```diff
@@ -0,0 +1,184 @@
+   * The amount of stable tokens being bought/sold in the exchange is required to be within a governable range.
+   * The assets being sold in the exchange are deposited.
+   * The current oracle price for the exchange is recorded.
+2. The proposed exchange must be approved by a multisig that has previously been authorized by Governance. At this point, the exchange still cannot be executed.
+3. A forced waiting period of X days must elapse before the exchange can be executed. During this time, Governance can choose to veto the exchange, refunding the exchange proposer.
+4. After the waiting period, the exchange is executed.
+
+An exchange can have the following states:
+1. Proposed - The exchange has been proposed, but not yet approved by the approver. The proposer may still cancel their proposal and be refunded their deposit.
+2. Approved - The exchange has been proposed and approved by the approver, but not yet executed or vetoed. The proposer may not cancel their proposal.
+3. Executed - The exchange has been proposed, approved by the approver, and executed.
+4. Cancelled - The exchange has been cancelled by the proposer when it was in the Proposed state, or it has been vetoed by Governance when it was in the Approved stage.
+
+### Specific Changes:
+
+#### Modifications of existing contracts:
+
+##### `StableToken.sol`
+
+Currently, `StableToken.sol` only allows its `Exchange.sol` or `Validators.sol` to [mint](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/StableToken.sol#L225) new stable tokens. Similarly, only its `Exchange.sol` can [burn](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/StableToken.sol#L275) stable tokens.
+
+The following modifications will be made:
+1. **Modification to [mint()](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/StableToken.sol#L224)**
+   * Allow `msg.sender` to be the stable token's `GrandaMento` in addition to the existing permitted senders, Exchange and `Validators.sol`.
+2. **Modification to [burn()](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/StableToken.sol#L273)**
+   * Allow `msg.sender` to be the stable token's `GrandaMento` in addition to the existing permitted sender, the Exchange.
+
+#### New contracts:
+
+##### `GrandaMento.sol`
+
+A new contract, `GrandaMento.sol`, is created. One deployment of `GrandaMento` exists per stable token. It must have the ability to mint/burn the corresponding stable token and be a spender of `Reserve.sol`.
+
+The contract is owned Governance, is freezable, and has the following configurable parameters:
+
+1. **`address public approver;`** - Set by governance. Intended to be a multisig and has the authority to approve exchanges.
+2. **`uint256 public minStableExchangeAmount;`** - Set by governance. Minimum amount of the stable token an exchange can buy/sell.
```

Comment:
Should we propose what a min/max exchange amount will be when we init this contract? Let's say 200k - 10M?
200k is a bit more than what we've seen people try to (and successfully) convert and make it seem accessible to individuals. 10M is the upper cap of what I've heard *may* we attempted at the very upper end of possibilities. 5M is more realistically what may be done for the future UBI program, so would be a more conservative number.

Reply:
Wonder if we should leave this decision to the CGP and leave the CIP as a specification, wdyt?

Reply:
Sounds good. Let's keep these parameters in mind for the CGP.